### PR TITLE
Geolocation filtering

### PR DESCRIPTION
**Pull request by petschki**
refactored from collective.venue ... open discussion

### CODE REVIEW

File: src/collective/collectionfilter/resources/collectionfilter.js [hunk 1]
```diff
@@ -108,6 +110,23 @@ define([
                 this.reloadCollection(collectionURL);
             }.bind(this));
 
+            // OPTION4 - maps filter
+            if (this.$el.hasClass('collectionMaps')) {
+                $('.pat-leaflet', this.$el).on('moveend zoomend', function (e) {
```
Comment (petschki) [inline src/collective/collectionfilter/resources/collectionfilter.js]: this event is not fired ... any hints how to get L.map.on('zoomend') here?

Comment (thet) [reply src/collective/collectionfilter/resources/collectionfilter.js]: Good catch :) This should do it: https://github.com/Patternslib/pat-leaflet/pull/3

**Comment by thet** [comment]
In general: +1 for that. Didn't have the time to review yet, though.
I wasn't sure about the package structure of https://github.com/collective/collective.venue/pull/13 anyways - I more or less added to collective.venue because this is where the map integration actually happens.
But I like your approach better.

File: src/collective/collectionfilter/resources/collectionfilter.js (hunk 1 above)
Comment (petschki) [reply src/collective/collectionfilter/resources/collectionfilter.js]: I tried this pull request and it works like a charm! though I need to get somehow `map.getBounds()` of the new view area and I could not find any map object neither on the `original_event` nor on the node ... do you have an idea how to get the new boundaries after zooming or moving?

**Comment by petschki** [comment]
@thet thank you for the detailed review. Funny because yesterday I also thought that the map functionality isn‘t really the correct place here. 

So I‘m  +1 for nearly all your comments. 

Map config within the tile was my idea because this makes it more flexible for the Editor to use different Maps in different contexts. Couldn‘t the API key be defined portal wide and the Maps Layout be configured locally? I like the flexibility in general and let the editor decide about consistency.

So I would suggest to bring the Maps logic to `plone.formwidget.geolocation` because other forms could then also benefit from this. If you agree I‘ll move the stuff over there and make a PR. 

The index makes sense on the behavior. I‘d move it over there.

**Comment by thet** [comment]
Thanks for your work on this!
For me, to use this refactored features I have to get it feature complete to the original collective.venue approach. Basically what is missing is filtering for events which the ILocatation behavior from collective.venue.
I'm working on this but as long I haven't finished it I'll not merge this. However, should be done within a week.

**Comment by petschki** [comment]
@thet no problem. I already use this branch in a new project of us and it works quite well. But I also need some more features like different colors of the map pins depending on some state of the article, but that belongs to `plone.patternslib` and `pat-leaflet`.

Not to forget, that this PR needs https://github.com/collective/plone.formwidget.geolocation/pull/12 and https://github.com/collective/collective.geolocationbehavior/pull/6 too ... I suggest releasing everything together and provide a `versions.cfg` here for the geolocation dependencies (like mosaic does for it's dependencies) which you can extend from in `buildout.cfg`

**Comment by thet** [comment]
@petschki you might be interestend in: https://github.com/Patternslib/pat-leaflet/commit/4828026639f9721a96f9d108d49b88b0362fcb60 eventually also https://github.com/Patternslib/pat-leaflet/commit/d342a65871a72f07262b98e2cc665f8480ab4ed2
looking at it, i should have named ``color`` rather ``pin_color``.
it's already merged into plone.patternslib

**Comment by thet** [comment]
@petschki i'm currently working on all the collectionfilter/maps branches.

**Pull request merged**